File: examples/arduino/arduino.js
var Cylon = require('cylon');

Cylon.config({
  api: {
    ssl: false,
    port: '8080',
    host: '0.0.0.0'
  }
});

Cylon.api();

Cylon.robot({
  name: 'arduino',
  connection: { name: 'arduino', adaptor: 'firmata', port: '/dev/ttyACM0'  },
  device: {name: 'led', driver: 'led', pin: 13}
}).start();

File: examples/arduino/arduino.go
package main

import (
	"github.com/hybridgroup/gobot"
	"github.com/hybridgroup/gobot/api"
	"github.com/hybridgroup/gobot/platforms/firmata"
	"github.com/hybridgroup/gobot/platforms/gpio"
)

func main() {
	gbot := gobot.NewGobot()
	api := api.NewAPI(gbot)
	adaptor := firmata.NewFirmataAdaptor("arduino", "/dev/ttyACM0")
	led := gpio.NewLedDriver(adaptor, "led", "13")

	api.Port = "8080"

	robot := gobot.NewRobot("arduino",
		[]gobot.Connection{adaptor},
		[]gobot.Device{led},
		func() {},
	)

	led.AddCommand("turn_on", func(params map[string]interface{}) interface{} {
		return led.On()
	})

	led.AddCommand("turn_off", func(params map[string]interface{}) interface{} {
		return led.Off()
	})

	gbot.AddRobot(robot)
	api.Start()
	gbot.Start()
}
